src/form/Form.d.ts

```diff
@@ -0,0 +1,118 @@
+/// <reference types="react" />
+import { FieldValues, setValues } from '../utils/init';
+
+export interface ElementProps {
+  [fieldId: string]: {
+    [propName: string]: string;
```

Comment:
The property can also be an Object type (specifically documented for a style but possibly other properties).  Not sure how deep this could go.